bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeBinding.java

```diff
@@ -369,28 +339,19 @@ private void readEcobee(OAuthCredentials oauthCredentials, Selection selection)
 		// Iterate through bindings and update all inbound values.
 		for (final EcobeeBindingProvider provider : this.providers) {
 			for (final String itemName : provider.getItemNames()) {
-				if (provider.isInBound(itemName)) {
+				if (provider.isInBound(itemName)
+						&& oauthCredentials == getOAuthCredentials(provider.getUserid(itemName))) {
```

Comment:
Since the objects come from the same map, I am actually only interested if they are the same object, not equal in content.  If they were equal in content but different references, it would be a bug.  Perhaps there is a better way to do this.  This line is a mirror of line 711.